react/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,35 +1,33 @@
+import { createTypes, completeTypes, withPostSuccess } from 'redux-recompose';
+
 import dataFetch from '../../services/MatchesService';
 
-export const actions = {
-  SET_AUTH: '@@DATA/SET_AUTH',
-  LOGIN_REQUEST: '@@DATA/LOGIN_REQUEST',
-  LOGIN_SUCCESS: '@@DATA/GLOGIN_SUCCESS',
-  LOGIN_FAILURE: '@@DATA/LOGIN_FAILURE'
-};
+export const actions = createTypes(completeTypes(['ON_LOGIN', 'SET_AUTH', 'ON_LOGOUT']), '@@SOCCER');
```

Comment:
i dont think `SET_AUTH` and `ON_LOGOUT` need the complete types, remember that `completeTypes` creates the actions `SET_AUTH_FAILURE` and `SET_AUTH_SUCCESS` to handle the loading state, and set auth and loading dont seem to need a loading state.